token/transfer-hook/cli/src/main.rs

```diff
@@ -0,0 +1,405 @@
+                .arg(
+                    Arg::with_name("transfer_hook_account")
+                        .value_parser(parse_transfer_hook_account)
+                        .value_name("PUBKEY:ROLE")
+                        .takes_value(true)
+                        .multiple(true)
+                        .min_values(0)
+                        .index(3)
+                        .help("Additional pubkey(s) required for a transfer hook and their \
+                            role, in the format \"<PUBKEY>:<ROLE>\". The role must be \
+                            \"readonly\", \"writable\". \"readonly-signer\", or \"writable-signer\".")
+                )
+                .arg(
+                    Arg::new("mint_authority")
+                        .long("mint-authority")
+                        .value_name("KEYPAIR")
+                        .validator(|s| is_valid_signer(s))
+                        .takes_value(true)
+                        .global(true)
+                        .help("Filepath or URL to mint-authority keypair [default: client keypair]"),
+                )
+        ).get_matches();
+
+    let (command, matches) = app_matches.subcommand().unwrap();
+    let mut wallet_manager: Option<Rc<RemoteWalletManager>> = None;
+
+    let cli_config = if let Some(config_file) = matches.value_of("config_file") {
+        solana_cli_config::Config::load(config_file).unwrap_or_default()
+    } else {
+        solana_cli_config::Config::default()
+    };
+
+    let config = {
+        let default_signer = DefaultSigner::new(
+            "fee_payer",
+            matches
+                .value_of("fee_payer")
+                .map(|s| s.to_string())
+                .unwrap_or_else(|| cli_config.keypair_path.clone()),
+        );
+
+        let json_rpc_url = normalize_to_url_if_moniker(
+            matches
+                .value_of("json_rpc_url")
+                .unwrap_or(&cli_config.json_rpc_url),
+        );
+
+        Config {
+            commitment_config: CommitmentConfig::confirmed(),
+            default_signer: default_signer
+                .signer_from_path(matches, &mut wallet_manager)
+                .unwrap_or_else(|err| {
+                    eprintln!("error: {err}");
+                    exit(1);
+                }),
+            json_rpc_url,
+            verbose: matches.is_present("verbose"),
+        }
+    };
+    solana_logger::setup_with_default("solana=info");
+
+    if config.verbose {
+        println!("JSON RPC URL: {}", config.json_rpc_url);
+    }
+    let rpc_client =
+        RpcClient::new_with_commitment(config.json_rpc_url.clone(), config.commitment_config);
+
+    match (command, matches) {
+        ("create-extra-account-metas", arg_matches) => {
+            let program_id = pubkey_of_signer(arg_matches, "program_id", &mut wallet_manager)
+                .unwrap()
+                .unwrap();
+            let token = pubkey_of_signer(arg_matches, "token", &mut wallet_manager)
```

Comment:
Also curious why this one involves signer/keypair. Maybe I'm not understanding how these methods work, but they appear to attempt to find the keypair at some path.

Reply:
It's more of a convenience, allowing me to specify `mint.json` on the command line instead of only its address

Reply:
Yep that's the idea! I'm pretty lazy when I write shell scripts and just want to use keypair files haha